sourcegraph/sourcegraph
# JetBrains: remove non-agent autocomplete and chat

Previously, the plugin had two code paths for autocomplete and chat: agent and non-agent. The non-agent path is no longer supported and can therefore be removed.

The motivation to remove this code is to simplify troubleshooting issues. Previously, the plugin would occasionally fallback to the non-agent code paths if the agent was not running. Now, we only try the agent instead.

This change makes it easier to troubleshoot issues since the logs will no longer include exceptions from non-agent code paths that were a red herring. This change also makes it easier to improve error handling since we don't have a fallback method we silently start using when the agent fails. Instead, we have to deal with the agent connection error immediately or report a helpful error message.

## Test plan

* Pull latest Cody changes
* `./gradlew -PforceAgentBuild=true :runIDE`
* Make sure autocomplete works
* Make sure sending a chat message works. Note that the embedding status is now hardcoded to an empty git repository.
